### ISSUE TITLE
Provide support for earlier shader versions

### ISSUE THREAD

**Issue by nixfreak** [treeform/shady#7]
Anyway you can provide earlier versions of shader to be used instead of the latest? 

For instance #version 330 

`0:1(10): error: GLSL 4.10 is not supported. Supported versions are: 1.10, 1.20, 1.30, 1.40, 1.50, 3.30, 1.00 ES, and 3.00 ES
`

**Comment by treeform** [comment]
Yes you can pass earlier version to the toGLSL macro:

```nim
macro toGLSL*(
  s: typed,
  version = "410",
  extra = "precision highp float;\n"
): string =
```

I have not tested with 3xx versions so I don't know if all features will work. Basic features? Yes. Advanced? I don't think so...

**Comment by nixfreak** [comment]
Are you saying you can change the version in the macro in one of the examples i.e. flare?

**Comment by treeform** [comment]
Yes. But if your code uses more advanced features it will not work.
